test/ledger.js

```diff
@@ -0,0 +1,123 @@
+    {"ledger":{"accepted":true,"account_hash":"A21ED30C04C88046FC61DB9DC19375EEDBD365FD8C17286F27127DF804E9CAA6","close_time_resolution":30,"closed":true,"ledger_index":"2","seqNum":"2","totalCoins":"100000000000000000","total_coins":"100000000000000000","transaction_hash":"0000000000000000000000000000000000000000000000000000000000000000"},"ledger_index":2,"validated":true},
+    {"ledger":{"closed":false,"ledger_index":"3","seqNum":"3"},"ledger_current_index":3,"validated":false},
+    {"error":"remoteError","error_message":"Remote reported an error.","remote":{"error":"lgrNotFound","error_code":20,"error_message":"ledgerNotFound","id":4,"request":{"command":"ledger","id":4,"ledger_index":4},"status":"error","type":"response"}}
+  ];
+
+  var specialIndexToExpected = {
+    "validated": 1,
+    "closed": 1,
+    "current": 2
+  };
+
+  var filterFields = function (ledger) {
+    if(typeof ledger === 'object') {
+      ledger = lodash.omit(ledger, ['close_time', 'close_time_human', 'hash', 'ledger_hash', 'parent_hash']);
+      Object.keys(ledger).map(function(key) {
+        ledger[key] = filterFields(ledger[key]);
+      });
+    }
+    return ledger;
+  }
+
+  var goodLedger = function (index, self, callback) {
+    self.what = "Good Ledger " + index;
+
+    var request = $.remote.request_ledger(index)
+      .on('success', function(ledger) {
+        // console.log("Ledger: %s", JSON.stringify(ledger));
+        ledger = filterFields(ledger);
+        // console.log("Ledger: %s", JSON.stringify(ledger));
+
+        var expected = specialIndexToExpected[index];
+        if(typeof expected === 'undefined')
+          expected = index-1;
+        // console.log("Index: %s, Expected: %s", index, expected);
+        assert.deepEqual(ledger, expectedledgers[expected]);
+
+        callback();
+      })
+      .on('error', function(ledger) {
+        assert(false, self.what);
+      })
+      .request();
+  };
+
+  var badLedger = function (index, self, callback) {
+    self.what = "Bad Ledger " + index;
+
+    var request = $.remote.request_ledger(index)
+      .on('success', function(ledger) {
+        assert(false, self.what);
+      })
+      .on('error', function(ledger) {
+        // console.log("Error: %s", JSON.stringify(ledger));
+        assert.deepEqual(ledger, expectedledgers[index-1]);
+
+        callback();
+      })
+      .request();
+  };
+
+  setup(function(done) {
+    testutils.build_setup().call($, done);
+  });
+
+  teardown(function(done) {
+    testutils.build_teardown().call($, done);
+  });
+
+  test("get ledgers by index", function(done) {
+    var self = this;
+
+    async.waterfall([
+      function (callback) {
+        goodLedger(1, self, callback);
+      },
+      function (callback) {
+        goodLedger(2, self, callback);
+      },
+      function (callback) {
+        goodLedger(3, self, callback);
+      },
+      function (callback) {
+        badLedger(4, self, callback);
+      },
+      ], function(error) {
```

Comment:
Your indentation is weird here, probably because you aren't indenting that list any more than the actual parameters... (ditto below).